cmd/pilot-agent/main.go

```diff
@@ -118,6 +123,28 @@ var (
 			proxyConfig.ConnectTimeout = ptypes.DurationProto(connectTimeout)
 			proxyConfig.StatsdUdpAddress = statsdUDPAddress
 			proxyConfig.ProxyAdminPort = int32(proxyAdminPort)
+			proxyConfig.MixerAddress = mixerAddress
+			switch infraAuthPolicy {
+			case int(proxyconfig.AuthPolicy_NONE):
+				proxyConfig.InfraAuthPolicy = proxyconfig.AuthPolicy_NONE
+			case int(proxyconfig.AuthPolicy_MUTUAL_TLS):
+				var ns string
+				proxyConfig.InfraAuthPolicy = proxyconfig.AuthPolicy_MUTUAL_TLS
+				if serviceregistry == platform.KubernetesRegistry {
+					partNS := strings.Split(discoveryAddress, ".")
+					if len(partNS) > 1 {
+						parts := strings.Split(partNS[1], ":")
+						if len(parts) > 0 && len(parts[0]) > 0 {
+							ns = parts[0]
+						} else {
+							ns = os.Getenv("POD_NAMESPACE")
+						}
+					} else {
+						ns = os.Getenv("POD_NAMESPACE")
+					}
+				}
+				proxyConfig.PilotSan = envoy.GetPilotSAN(pilotDomain, ns)
```

Comment:
Few comments ? I assume you want the service account of pilot ? Can we have a default value (matching what we have in the default config ), and a flag to override ?

Reply:
It would be really great if we could add more comments here. The code review UI is not very user-friendly.